pkg/etcd/etcd.go

```diff
@@ -916,6 +950,97 @@ func (e *ETCD) listSnapshots(ctx context.Context, snapshotDir string) ([]snapsho
 	return snapshots, nil
 }
 
+// updateConfigMap
+func updateConfigMap(data map[string]string, snapshotFiles []snapshotFile) error {
+	for _, v := range snapshotFiles {
+		b, err := json.Marshal(v)
+		if err != nil {
+			return err
+		}
+		data[v.Name] = string(b)
+	}
+
+	return nil
+}
+
+// storeSnapshotData stores the given snapshot data in the "snapshots" ConfigMap.
+func (e *ETCD) storeSnapshotData(ctx context.Context, snapshotFiles []snapshotFile) error {
+	snapshotConfigMap, err := e.cs.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(ctx, "snapshots", metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			data := make(map[string]string, len(snapshotFiles))
+			if err := updateConfigMap(data, snapshotFiles); err != nil {
+				return err
+			}
+			cm := v1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ConfigMap",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "snapshots",
```

Comment:
```suggestion
					Name:      version.Program + "-etcd-snapshots",
```
but also make it a variable since it's repeated down below in the error handlers